pyracf/resource/resource_admin.py

```diff
@@ -265,11 +265,316 @@ def set_universal_access(
         )
         return self._to_steps(result)
 
+    # ============================================================================
+    # Individual Access
+    # ============================================================================
     def get_my_access(self, resource: str, class_name: str) -> Union[str, bytes, None]:
         """Get the access associated with your own general resource profile."""
         profile = self.extract(resource, class_name, profile_only=True)
         return self._get_field(profile, "base", "yourAccess")
 
+    # ============================================================================
+    # Auditing Rules
+    # ============================================================================
+    def get_audit_rules(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes, None]:
+        """Get the auditing rules associated with this general resource profile."""
+        profile = self.extract(resource, class_name, profile_only=True)
+        return self._get_field(profile, "base", "auditing")
+
+    def clear_audit_failures(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes]:
+        """
+        Clears the auditing rules for failures for this general resource profile while
+        preserving other rules.
+        """
+        result = [self.extract(resource, class_name)]
+        profile = result[0]["securityResult"]["resource"]["commands"][0]["profiles"][0]
+        audit_rules = self._get_field(profile, "base", "auditing")
+        if "failures" not in audit_rules:
+            return self._to_steps(result[0])
+        traits = {}
+        if "success" in audit_rules:
+            traits[f"base:audit_{audit_rules['success']}"] = "SUCCESS"
+        if "all" in audit_rules:
+            traits[f"base:audit_{audit_rules['all']}"] = "ALL"
```

Comment:
If there is currently an audit rule for failures shouldn't there be a trait that unsets that, or is it that anytime that you set any audit rules, all existing audit rules will be overwritten with what you specify?

Reply:
I think this is covered by our discussion of the functionality and re-design of these functions. Let me know if this is still a concern.

Reply:
Ok, so based on our conversation on this functionality, my understanding of how overwrite works is that we don't need to explicitly clear any audit rules because the act of creating new audit rules deletes any audit rules that exist, which makes this question irrelevant.